packages/gallery/tests/styleParams/arrowsVerticalPosition.spec.js

```diff
@@ -0,0 +1,86 @@
+  let driver;
+  const initialProps = {
+    container,
+    items: images2,
+    options,
+  };
+
+  beforeEach(() => {
+    driver = new GalleryDriver();
+    //base layout styles for entire test-suite
+    initialProps.options = mergeNestedObjects(initialProps.options, {
+      galleryLayout: GALLERY_CONSTS.layout.SLIDESHOW,
+      slideshowInfoSize: 39,
+    });
+  });
+
+  afterEach(() => {
+    driver.detach.proGallery();
+  });
+
+  async function mountGallery(props) {
+    driver.mount.proGallery(props);
+    await driver.update();
+  }
+
+  async function mountGalleryAndGetArrows(props) {
+    await mountGallery(props); // mounting gallery
+    return driver.find.selector('.nav-arrows-container'); // Getting the arrows
+  }
+
+  it('Should not render the arrows when not enough info space', async () => {
+    // Exact style params relevant to this sub-test
+    initialProps.options = mergeNestedObjects(initialProps.options, {
+      slideshowInfoSize: 38, // info height is 39 when default, so slideshowInfoSize < arrowsSize
+      arrowsVerticalPosition: 'INFO_CENTER',
+    });
+    const navArrows = await mountGalleryAndGetArrows(initialProps);
+    expect(navArrows).to.have.lengthOf(0);
+  });
+
+  it('Checks if "INFO_CENTER" has correct distance from top for SlideShow', async () => {
+    // Exact style params relevant to this sub-test
+    initialProps.options = mergeNestedObjects(initialProps.options, {
+      arrowsVerticalPosition: 'INFO_CENTER',
+    });
+    const navArrows = await mountGalleryAndGetArrows(initialProps);
+    const galleryContainer = driver.find.selector('#pro-gallery-container');
+    const { height: galleryHeight } = galleryContainer.props().style;
+    const { top } = navArrows.props().style;
+    const expectedInfoSpace = (-1 * galleryHeight) / 2;
+    expect(top.replace(/\s/g, '')).to.eq(
+      `calc(50%+0px-19.5px-${expectedInfoSpace}px)`
+    );
+  });
+
+  it('Checks if "ITEM_CENTER" has correct distance from top for SlideShow', async () => {
+    // Exact style params relevant to this sub-test
+    initialProps.options = mergeNestedObjects(initialProps.options, {
+      arrowsVerticalPosition: 'ITEM_CENTER',
+    });
+    const navArrows = await mountGalleryAndGetArrows(initialProps);
+    const { top } = navArrows.props().style;
+    expect(top.replace(/\s/g, '')).to.eq(`calc(50%+0px-19.5px-0px)`);
+  });
+
+  it('Checks if "IMAGE_CENTER" has correct distance from top for SlideShow', async () => {
+    // Exact style params relevant to this sub-test
+    initialProps.options = mergeNestedObjects(initialProps.options, {
+      arrowsVerticalPosition: 'IMAGE_CENTER',
+    });
+    const navArrows = await mountGalleryAndGetArrows(initialProps);
+    const { top } = navArrows.props().style;
+    const expectedInfoSpace = initialProps.options.slideshowInfoSize / 2;
+    expect(top.replace(/\s/g, '')).to.eq(
+      `calc(50%+0px-19.5px-${expectedInfoSpace}px)`
```

Comment:
At the end of this refactoring this complicated `calc` should be written only one time in this file

Reply:
Done 
@nirnaor